diff -rNw -U 0 py/pack/pack.py py/pack/pack.py
--- py/pack/pack.py
+++ py/pack/pack.py
@@ -554 +554 @@
-def randomPeriPack(radius,initSize,rRelFuzz=0.0,memoizeDb=None,noPrint=False):
+def randomPeriPack(radius,initSize,rRelFuzz=0.0,memoizeDb=None,noPrint=False,seed=0):
@@ -574 +574 @@
-	sp.makeCloud(Vector3().Zero,O.cell.refSize,radius,rRelFuzz,-1,True)
+	sp.makeCloud(Vector3().Zero,O.cell.refSize,radius,rRelFuzz,-1,True,seed=seed)
